atlasaction/action.go

```diff
@@ -79,6 +79,107 @@ func MigrateApply(ctx context.Context, client *atlasexec.Client, act *githubacti
 	return nil
 }
 
+const (
+	StatePending  = "PENDING_USER"
+	StateApproved = "APPROVED"
+	StateAborted  = "ABORTED"
+	StateApplied  = "APPLIED"
+)
+
+// MigrateDown runs the GitHub Action for "ariga/atlas-action/migrate/down".
+func MigrateDown(ctx context.Context, client *atlasexec.Client, act *githubactions.Action) (err error) {
+	var vars atlasexec.Vars
+	if v := act.GetInput("vars"); v != "" {
+		if err := json.Unmarshal([]byte(v), &vars); err != nil {
+			return fmt.Errorf("failed to parse vars: %w", err)
+		}
+	}
+	var a uint64
+	if i := act.GetInput("amount"); i != "" {
+		a, err = strconv.ParseUint(act.GetInput("amount"), 10, 64)
+		if err != nil {
+			return fmt.Errorf("failed to parse amount: %w", err)
+		}
+	}
+	params := &atlasexec.MigrateDownParams{
+		URL:       act.GetInput("url"),
+		DevURL:    act.GetInput("dev-url"),
+		DirURL:    act.GetInput("dir"),
+		ConfigURL: act.GetInput("config"),
+		Env:       act.GetInput("env"),
+		ToVersion: act.GetInput("to-version"),
+		ToTag:     act.GetInput("to-tag"),
+		Amount:    a,
+		Context: &atlasexec.DeployRunContext{
+			TriggerType:    atlasexec.TriggerTypeGithubAction,
+			TriggerVersion: Version,
+		},
+		Vars: vars,
+	}
+	if a := act.GetInput("amount"); a != "" {
+		params.Amount, err = strconv.ParseUint(a, 10, 64)
+		if err != nil {
+			return err
+		}
+	}
+	// Based on the retry configuration values, retry the action if there is an error.
+	var (
+		started           = time.Now()
+		interval, timeout time.Duration
+	)
+	switch wi, wt := act.GetInput("wait-interval"), act.GetInput("wait-timeout"); {
+	case wi == "" && wt == "": // default, run once
+	case wi != "" && wt != "":
+		// Configure the wait / retry behaviour as defined by the caller.
+		wii, err := strconv.Atoi(wi)
+		if err != nil {
+			return fmt.Errorf(`parsing "wait-interval": %w`, err)
+		}
+		wti, err := strconv.Atoi(wt)
+		if err != nil {
+			return fmt.Errorf(`parsing "wait-timeout": %w`, err)
+		}
+		interval = time.Duration(wii) * time.Second
+		timeout = time.Duration(wti) * time.Second
```

Comment:
Maybe accept a duration value. It's more flexible and that's what used by (some of) Atlas flags